File: NetCoreApi2/clientApp/src/features/empresas/models/empresas.ts
import {ClientesModel} from "@/features/clientes/models/clientes.ts";

export interface EmpresasInterface {
    id: number;
    nombre: string;
    cif: string;
    clientes: ClientesModel[];
}//end interface

export class EmpresasModel{
    id: number;
    nombre: string;
    cif: string;
    clientes: ClientesModel[];

    constructor(data: EmpresasInterface){
        this.id = data.id;
        this.nombre = data.nombre;
        this.cif = data.cif;
        this.clientes = data.clientes;
    }//end constructor
}//end class EmpresasModel

File: NetCoreApi2/Data/Models/Empresas.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace NetCoreApi2.Data.Models
{
    public class EmpresasModel
    {
        [Key]
        public int Id { get; set; }
        public string nombre { get; set; }
        public string CIF { get; set; }
        public ICollection<ClientesModel> Clientes { get; set; } 
    }
}
